gemrb/core/GUI/GameControl.cpp

```diff
@@ -2209,7 +2209,18 @@ bool GameControl::OnMouseUp(const MouseEvent& me, unsigned short Mod)
 		
 		if (overContainer || overDoor || (overInfoPoint && overInfoPoint->Type==ST_TRAVEL && target_mode == TARGET_MODE_NONE)) {
 			// move to the object before trying to interact with it
-			CommandSelectedMovement(p, false, isDoubleClick);
+			//core->GetFirstSelectedPC(false)
+			Actor* selectedActor = core->GetFirstSelectedPC(false);
+			if (!selectedActor) {
+				//this could be a non-PC
+				selectedActor = core->GetGame()->selected[0];
+			}
```

Comment:
we are now seeing this code duplicated 3x. It's time to make it a function.

Reply:
done